src/odemis/acq/stitching/_registrar.py

```diff
@@ -14,18 +14,442 @@
+                    if self.tiles[i][j] != None:                      
+                        md_pos_ij = self.tiles[i][j].metadata[model.MD_POS]
+                        dist = numpy.sqrt((md_pos[0]-md_pos_ij[0])**2 + (md_pos[1]-md_pos_ij[1])**2)
+                        if dist < minDist:
+                            minDist = dist
+                            pos_prev = (i,j)
+                            md_pos_prev = self.tiles[i][j].metadata[model.MD_POS]
+            
+            # Insert new tile either to the right or to the bottom of the closest tile.
+            ver_diff = abs(md_pos[1]-md_pos_prev[1])
+            hor_diff = abs(md_pos[0]-md_pos_prev[0])
+            if ver_diff > hor_diff:
+                self.posY = pos_prev[0] + 1
+                self.posX = pos_prev[1]
+                if len(self.shifts) <= self.posY:
+                    self._updateGrid("y") # extend grid in y direction
+            else:
+                self.posY = pos_prev[0]
+                self.posX = pos_prev[1] + 1    
+                if len(self.shifts[0]) <= self.posX:
+                    self._updateGrid("x") # extend grid in x direction
+     
+               
+            # Calculate overlap. The overlap is assumed to be identical for all tiles and is
+            # only calculated after the second tile is added.
+            if len(self.shifts)==2 and len(self.shifts[0])==1:
+                size_meter = 2*self.tiles[0][0].metadata[model.MD_POS][1]
+                diff = abs(tile.metadata[model.MD_POS][1] - self.tiles[0][0].metadata[model.MD_POS][1])
+                self.ovrlp = 1-(abs(size_meter-diff)/size_meter)
+            
+            elif len(self.shifts)==1 and len(self.shifts[0])==2:
+                size_meter = 2*self.tiles[0][0].metadata[model.MD_POS][0]
+                
+                diff = abs(tile.metadata[model.MD_POS][0] - self.tiles[0][0].metadata[model.MD_POS][0])
+                self.ovrlp = numpy.round(abs(size_meter-diff)/size_meter,decimals=2)
+      
+        self.osize = self.size * self.ovrlp  # overlap size in pixels  
+        self.tsize = int(self.size - self.osize)  
+        
+        self._stitch(tile,self.posY,self.posX)              
+             
+    def getPositions(self):
+        """ 
+        returns:
+        tile_positions (list of tuples): the adjusted position in X/Y for each tile, in the order they were added
+        dep_tile_positions (list of tuples of tuples): the adjusted position for each dependent tile 
+        (in the order they were passed)
+        """   
+        firstPosition = numpy.divide(self.tiles[0][0].metadata[model.MD_POS],self.px_size)
+        tile_positions = []
+        for i in range(len(self.acqOrder)):
+            shift = self.shifts[self.acqOrder[i][0]][self.acqOrder[i][1]]
+            tile_positions.append(((shift[0]+firstPosition[0])*self.px_size[0],(firstPosition[1]-shift[1])*self.px_size[1]))
+        # Return same positions for dependent tiles
```

Comment:
Computing the position of dependent tiles is simple, but not _that_ simple. The dependent tiles don't have the same position as the "main" tile. They have a fixed relative position compared to the main tile. So, in addTile(), you should store the shift between the dependent tile and the main tile, and here, add it back to the main tile position to report the dependent tiles positions.